src/java.base/share/classes/java/text/ListFormat.java

```diff
@@ -0,0 +1,571 @@
+        }
+        startPattern = Pattern.compile(startBefore + "(.+?)" + startBetween);
+        endPattern = Pattern.compile(endBetween + "(.+?)" + endAfter);
+    }
+
+    /**
+     * {@return the available locales that support list formatting}
+     */
+    public static Locale[] getAvailableLocales() {
+        // Same as a typical format class
+        return DateFormat.getAvailableLocales();
+    }
+
+    /**
+     * {@return the list format object for the default
+     * {@link Locale.Category#FORMAT FORMAT Locale}, {@code STANDARD} type,
+     * and {@code FULL} style}
+     */
+    public static ListFormat getInstance() {
+        return getInstance(Locale.getDefault(Locale.Category.FORMAT), Type.STANDARD, Style.FULL);
+    }
+
+    /**
+     * {@return the list format object for the specified {@code Locale}, {@code Type},
+     * and {@code Style}}
+     * @param locale Locale to be used, not null
+     * @param type type of the list format. One of STANDARD/OR/UNIT, not null
+     * @param style style of the list format. One of FULL/SHORT/NARROW, not null
+     * @throws NullPointerException if any of the arguments are null
+     */
+    public static ListFormat getInstance(Locale locale, Type type, Style style) {
+        Objects.requireNonNull(locale);
+        Objects.requireNonNull(type);
+        Objects.requireNonNull(style);
+        return new ListFormat(locale, LocaleProviderAdapter.forType(LocaleProviderAdapter.Type.CLDR)
+                .getLocaleResources(locale)
+                .getListPatterns(type, style));
+    }
+
+    /**
+     * {@return the list format for the specified patterns}
+     * <p>
+     * This factory returns an instance based on the customized patterns array,
+     * instead of letting the runtime provide appropriate patterns for the Locale/Type/Style.
+     * <p>
+     * The patterns array should contain five String patterns, each corresponding to the Unicode LDML's
+     * {@code listPatternPart}, i.e., "start", "middle", "end", two element, and three element patterns
+     * in this order. Each pattern contains "{0}" and "{1}" (and "{2}" for the three element pattern)
+     * placeholders that are substituted with the passed input strings on formatting.
+     * If the length of the patterns array is not 5, an {@code IllegalArgumentException}
+     * is thrown.
+     * <p>
+     * Each pattern string is first parsed as follows. Patterns in parentheses are optional:
+     * <blockquote><pre>
+     * start := (start_before){0}start_between{1}
+     * middle := {0}middle_between{1}
+     * end := {0}end_between{1}(end_after)
+     * two := (two_before){0}two_between{1}(two_after)
+     * three := (three_before){0}three_between{1}three_between{2}(three_after)
+     * </pre></blockquote>
+     * If parsing of the pattern string for start/middle/end fails, it throws an
+     * {@code IllegalArgumentException}. If two/three pattern string is empty, or
+     * fails on parsing, it falls back to
```

Comment:
I don't see code to try to parse the TWO or THREE patterns, where it could detect incorrectly formatted (user supplied) patterns.
If I understand correctly, a bad TWO or THREE pattern would not be detected until it is used in format or parse.

Reply:
Right. Will provide validation for them